mobile/src/main/java/ch/epfl/sdp/mobile/ui/ar/ArScreen.kt

```diff
@@ -32,42 +60,107 @@ fun ArScreen(modifier: Modifier = Modifier) {
   ): ArModelNode {
     val node =
         ArModelNode(placementMode = placementMode, autoAnchor = true, placementPosition = position)
-    val instance =
-        node.loadModel(
-            context = context,
-            glbFileLocation = glbPath,
-            autoScale = true,
-        )
-    val boundingBox = instance?.filamentAsset?.boundingBox
-    val halfExtent = boundingBox?.halfExtent?.maxOrNull()!!
-    // TODO : Determine the scale
-    node.scale(0.5f / halfExtent)
+    node.loadModel(
+        context = context,
+        glbFileLocation = glbPath,
+        autoScale = true,
+    )
     return node
   }
 
-  /**
-   * Add the given [ArModelNode] to the scene
-   *
-   * @param modelNode the model that we want to add
-   * @param view Where the model will be added
-   */
-  fun addNode(modelNode: ArModelNode?, view: ArSceneView) {
-    val node = modelNode ?: return
-    if (node !in view.children) {
-      view.addChild(node)
-    }
+  // Keep the screen only for this composable
+  DisposableEffect(Unit) {
```

Comment:
It's very unlikely (probably not even possible) that the `LocalView` changes but you should run the `DisposableEffect` again if it does.

```suggestion
  DisposableEffect(view) {
```

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/179/commits/198dc6e73229ae54fbf8dffde73c55071fb8ca03